src/PgNamed.hs

```diff
@@ -295,3 +300,12 @@ withNamedArgs qNamed namedArgs = do
         Right r      -> pure r
     args <- namesToRow names namedArgs
     pure (q, args)
+
+
+handleIO :: (MonadIO m, WithNamedError m) => IO a -> m a
+handleIO io = do
```

Comment:
Nitpick: I'd rather call this something more specific like `handleSqlError`

Reply:
keep in mind this also lifts IO to MonadIO m. I used handle as it's also function from `Control.Exception` (flip catch) as it sounded better than `catchIO`. With this in mind how about calling this `handlePgIO` ?

Reply:
I'd like to see the word exception/error in function name for clarity, so `handleIO`/`handlePgIO` don't cut it for me :-)
But it doesn't matter much, since it's not exposed anyway. How about rethrowSqlError?

Reply:
good point. well in my mind `handle` already refers to exception handling but I agree it's not that explicit. We can change this to `rethrowSqlError` to be explicit.